storybookjs/storybook
# Add CSF 2→3 banner to docs code snippets

Also added an "Upgrading from CSF 2 to CSF 3 section to [`/docs/api/csf`](https://storybook.js.org/docs/react/api/csf), which is largely based on https://storybook.js.org/blog/component-story-format-3-0/#new-in-csf-3-0.

See corresponding `frontpage` PR: https://github.com/storybookjs/frontpage/pull/437

**Preview:**
![Screenshot of code snippet banner pointing to upgrade guide and corresponding CSF 2 example](https://user-images.githubusercontent.com/486540/205178561-1234f185-8700-4a9f-9022-f7198947e726.png)

## How to test

To test it locally:

1. Check out the `kylegach/dx-392-build-csf-2--csf-3-snippet-banner` branch in the https://github.com/storybookjs/frontpage repo
1. In `frontpage`
	1. Run `yarn extract-monorepo-docs add-csf-2-to-3-banner`
	2. Run `GATSBY_SKIP_ADDON_PAGES=true yarn start`

That will pull down this branch's docs into your local `frontpage` and run the site locally. The local build won't have all of the redirects, so start here to avoid 404s: http://localhost:8000/docs/react/why-storybook.
